### ISSUE TITLE
Wacom HID 527F Finger touch not recognized as Touchscreen

### ISSUE THREAD

**Issue by TobiasBaeumlin** [xournalpp/xournalpp#3512]
OS: Fedora 34
Hardware: Lenovo X1 Yoga gen 6
Version: 1.1.0
Installation method: Both Fedora repos and flatpak

xinput shows:
⎡ Virtual core pointer                          id=2    [master pointer  (3)]
⎜   ↳ Virtual core XTEST pointer                id=4    [slave  pointer  (2)]
⎜   ↳ Lenovo Laser Wireless Mouse               id=11   [slave  pointer  (2)]
⎜   ↳ ELAN0672:00 04F3:3187 Mouse               id=12   [slave  pointer  (2)]
⎜   ↳ ELAN0672:00 04F3:3187 Touchpad            id=13   [slave  pointer  (2)]
⎜   ↳ Wacom HID 527F Pen stylus                 id=14   [slave  pointer  (2)]
⎜   ↳ Wacom HID 527F Finger touch               id=15   [slave  pointer  (2)]
⎜   ↳ TPPS/2 Elan TrackPoint                    id=19   [slave  pointer  (2)]
⎜   ↳ Wacom HID 527F Pen eraser                 id=21   [slave  pointer  (2)]
⎜   ↳ Microsoft Microsoft® 2.4GHz Transceiver v9.0 Consumer Control     id=24   [slave  pointer  (2)]
⎜   ↳ Microsoft Microsoft® 2.4GHz Transceiver v9.0 Mouse        id=26   [slave  pointer  (2)]
⎜   ↳ Microsoft Microsoft® 2.4GHz Transceiver v9.0 Consumer Control     id=27   [slave  pointer  (2)]
⎣ Virtual core keyboard                         id=3    [master keyboard (2)]

Under 'Input System' in Preferences none of the Wacom HID 527F devices is listed.

Using the pen and eraser works as expected, using fingers acts identical as pen.

Pen input is lagging with the version from Fedora repos and much more fluid with the flatpak.

**Comment by bhennion** [comment]
Could you give us a screenshot of your Input System settings window?

Also, what's your desktop environment (Gnome, KDE, etc) and is it running on X11 or Wayland?

**Comment by TobiasBaeumlin** [comment]
Here's the screenshot of the Input System settings window:

![Screenshot_20211105_125746](https://user-images.githubusercontent.com/840090/140507008-5a4f53bc-2bec-4108-8e3c-a8d0b98db2a6.png)

I'm using KDE running X11.

**Comment by personalizedrefrigerator** [comment]
Do you experience the same issue on Wayland?
What happens if you disable all devices (in the Xournal++ settings) except for the Wacom Pen?
All except for the Wacom touchscreen?
Do you have touchscreen drawing turned on?

My current guess is that GTK thinks your Wacom tablet is a trackpad:
![GTK parenthetical states that GTK thinks the device is a trackpad.](https://user-images.githubusercontent.com/46334387/140854036-418e5c89-1f14-4ca1-a3fa-8912a3e315e4.jpeg)

If this is the case (and the device is assigned to Touchscreen), then dragging the view with the touchscreen _should_ still work. This is why I'm asking the above questions.

**Comment by TobiasBaeumlin** [comment]
I just gave Wayland another try (which is a pain in the ... in other respects) and found that it doesn't have the same issue.
Also in Wayland the Wacom HID 527F devices are listed in the preferences.

I played around with devices in the Xournal++ settings with the strange result that they don't seem to matter at all. The behaviour of the input system remains the same, even if I disable all devices. Which strikes me as odd.

And no, I don't have any other Wacom devices attached to my computer.
I also noticed that in the touchscreen section of the preferences two of the Wacom HID 527F device are listed (pen and eraser) but the touchscreen is missing. See screenshot.

![Screenshot_20211109_122743](https://user-images.githubusercontent.com/840090/140916463-f904b671-018d-4300-bfa0-c98efa7cdf9f.png)
